jupyterhub/templates/hub/deployment.yaml

```diff
@@ -59,15 +59,8 @@ spec:
       {{- end }}
       securityContext:
         fsGroup: {{ .Values.hub.fsGid }}
-      {{- if or .Values.hub.imagePullSecret.enabled .Values.hub.image.pullSecrets }}
-      imagePullSecrets:
-        {{- if .Values.hub.imagePullSecret.enabled }}
-        - name: hub-image-credentials
-        {{ else }}
-        {{- range .Values.hub.image.pullSecrets }}
-        - name: {{ . }}
-        {{- end }}
-        {{- end }}
+      {{- with include "jupyterhub.imagePullSecrets" (dict "root" . "image" .Values.hub.image) }}
+      imagePullSecrets: {{ . }}
       {{- end }}
```

Comment:
I embedded the logic to render a JSON formatted output such as `["secret-name-a", "secret-name-b"]` or a falsy blank string in a template helper function to avoid adding this otherwise fluffy logic in each pod.